SPECS/numpy/numpy.spec

```
@@ -1,131 +1,857 @@
 Requires:       python3-devel
-Requires:       python3-numpy = %{version}-%{release}
+Requires:       python3-numpy%{?_isa} = %{version}-%{release}
 Provides:       python3-f2py = %{version}-%{release}
+Obsoletes:      python3-f2py <= 2.45.241_1927
+Provides:       f2py = %{version}-%{release}
+Provides:       numpy-f2py = %{version}-%{release}
+Obsoletes:      numpy-f2py < 1:1.10.1-3
 
 %description -n python3-numpy-f2py
 This package includes a version of f2py that works properly with NumPy.
 
+%package -n python3-numpy-doc
+Summary:        Documentation for numpy
+License:        BSD AND Python AND ASL 2.0
+Requires:       python3-numpy = %{version}-%{release}
+BuildArch:      noarch
+
+%description -n python3-numpy-doc
+This package provides the complete documentation for NumPy.
+
 %prep
 %autosetup -p1
 
+# Force re-cythonization (ifed for PKG-INFO presence in setup.py)
+# rm PKG-INFO
+
+# openblas is provided by flexiblas by default; otherwise,
+# Use openblas pthreads as recommended by upstream (see comment in site.cfg.example)
+cat >> site.cfg <<EOF
+[openblas]
+libraries = %{blaslib}%{blasvar}
+library_dirs = %{_libdir}
+EOF
+
 %build
-%py3_build
+%set_build_flags
+
+env OPENBLAS=%{_libdir} \
+    BLAS=%{_libdir} \
+    LAPACK=%{_libdir} CFLAGS="%{optflags}" \
+    %{__python3} setup.py build
 
-# using py3_install macro, numpy header files are missing. --skip-build arg is removed essentially with this change.
-# https://github.com/numpy/numpy/issues/16005
-# changes imported from Fedora.
 %install
-%{__python3} setup.py install --root %{buildroot}
+mkdir docs
+pushd docs
+unzip %{SOURCE1}
+popd
+
+#%%{__python3} setup.py install -O1 --skip-build --root %%{buildroot}
+# skip-build currently broken, this works around it for now
+env OPENBLAS=%{_libdir} \
+    FFTW=%{_libdir} BLAS=%{_libdir} \
+    LAPACK=%{_libdir} CFLAGS="%{optflags}" \
+    %{__python3} setup.py install --root %{buildroot} --prefix=%{_prefix}
+pushd %{buildroot}%{_bindir} &> /dev/null
+ln -s f2py3 f2py.numpy
+popd &> /dev/null
+
+#symlink for includes, BZ 185079
+mkdir -p %{buildroot}%{_includedir}
+ln -s %{python3_sitearch}/%{name}/core/include/numpy/ %{buildroot}%{_includedir}/numpy
+
+# distutils from setuptools don't have the patch that was created to avoid standard runpath here
+# we strip it manually instead
+# ERROR   0001: file '...' contains a standard runpath '/usr/lib64' in [/usr/lib64]
+chrpath --delete %{buildroot}%{python3_sitearch}/%{name}/core/_multiarray_umath.*.so
+chrpath --delete %{buildroot}%{python3_sitearch}/%{name}/linalg/lapack_lite.*.so
+chrpath --delete %{buildroot}%{python3_sitearch}/%{name}/linalg/_umath_linalg.*.so
+
 
 %check
-pip3 install nose pytest
-mkdir -pv test
-cd test
-PYTHONPATH=%{buildroot}%{python3_sitelib} PATH=$PATH:%{buildroot}%{_bindir} %python3 -c "import numpy; numpy.test()"
+%if %{with_check}
+export PYTHONPATH=%{buildroot}%{python3_sitearch}
+pip install pytest hypothesis typing-extensions
```

We've had some issue with `pytest` breaking our builds, if allowed to install the latest version like this. Please take a look at the version currently downloaded for these tests and "freeze" it here like so:
```suggestion
pip install pytest==[some_version_number] hypothesis typing-extensions
```